src/core/tools/common/dsn_message_parser.cpp

```diff
@@ -113,7 +113,68 @@ namespace dsn
 
     int dsn_message_parser::prepare_on_send(message_ex* msg)
     {
-        return (int)msg->buffers.size();
+        auto& header = msg->header;
+        auto& buffers = msg->buffers;
+
+        if (task_spec::get(msg->local_rpc_code)->rpc_message_crc_required)
```

Comment:
This introduces performance issues as this function is called inside locks. Need to do this before linking the message into the sending queue.